lib/reek/context_builder.rb

```diff
@@ -101,93 +140,278 @@ def process_send(exp)
+    #
+    # while(x < 5)
+    #   puts 'bingo'
+    # end
+    #
+    # Counts the `while` body as one statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `children[1]` below refers to the `while` body (so `puts 'bingo'` from above)
+    #
     def process_while(exp)
       increase_statement_count_by(exp.children[1])
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
     alias_method :process_until, :process_while
 
+    # Handles `for` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # for i in [1,2,3,4]
+    #   puts i
+    # end
+    #
+    # Counts the `for` body as one statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `children[2]` below refers to the `while` body (so `puts i` from above)
+    #
     def process_for(exp)
       increase_statement_count_by(exp.children[2])
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
+    # Handles `rescue` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # def simple
+    #   begin
+    #     raise ArgumentError, 'raising...'
+    #   rescue => e
+    #   end
+    # end
+    #
+    # Counts the `rescue` body as one statement.
+    #
+    # At the end we subtract one statement because the surrounding context was already counted
+    # as one (e.g. via `process_def`).
+    #
+    # `exp.children.first` below refers to the everything that is covered by the `rescue`
+    # node body (which is everything from `begin` to `rescue` in the example above).
+    #
     def process_rescue(exp)
       increase_statement_count_by(exp.children.first)
       decrease_statement_count
-      process_default(exp)
+      process(exp)
     end
 
+    # Handles `resbody` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # def simple
+    #   raise ArgumentError, 'raising...'
+    #   rescue => e
+    #     puts 'rescued!'
+    #   end
+    # end
+    #
+    # Counts the exception capturing and every statement after that.
+    #
+    # So `exp.children[1..-1]` from the code below would be an array with the following 2 elements:
+    # [
+    #   (lvasgn :e),
+    #   (send nil :puts (str "rescued!"))
+    # ]
+    #
+    # which thus counts as 2 statements.
+    #
     def process_resbody(exp)
       increase_statement_count_by(exp.children[1..-1].compact)
-      process_default(exp)
+      process(exp)
     end
 
+    # Handles `case` nodes.
+    #
+    # An input example that would trigger this method would be:
+    #
+    # foo = 5
+    # case foo
+    #   when (1..100)
+    #     puts 'In between'
+    #   else
+    #     puts 'Not sure what I got here'
+    # end
```

Comment:
I’d indent `when` and `else` the same as `case` and `end`, and maybe drop the parentheses around `1..100`.